src/Symfony/Bundle/SecurityBundle/Command/UserPasswordEncoderCommand.php

```diff
@@ -32,10 +35,10 @@ protected function configure()
     {
         $this
             ->setName('security:encode-password')
-            ->setDescription('Encode a password.')
+            ->setDescription('Encodes a password.')
             ->addArgument('password', InputArgument::OPTIONAL, 'Enter a password')
-            ->addArgument('user-class', InputArgument::OPTIONAL, 'Enter the user class configured to find the encoder you need.')
-            ->addArgument('salt', InputArgument::OPTIONAL, 'Enter the salt you want to use to encode your password.')
+            ->addArgument('user-class', InputArgument::OPTIONAL, 'Enter the user class configured to find the encoder you need.', 'Symfony\Component\Security\Core\User\User')
+            ->addOption('empty-salt', null, InputOption::VALUE_NONE, 'Do not generate a salt or let the encoder generate one.')
```

Comment:
If we add the `--salt` option I don't think the command necessarily complicates:

  * If you know nothing about the salt, you won't use this option and the command won't ask you for it.
  * If you know what a salt is, but don't use this option, the command won't bother you and everything will work.
  * If you are an advanced user, read the command help, and nedd (for whatever reason) to use a custom salt, you just use this super easy `--salt` option. There is only one exception: if you are also using Bcrypt, the command will remind you that you cannot provide a salt in this case, will ignore it and everything will work as expected.